### PR TITLE
Additional refactors based on the PR #2206

### PR DESCRIPTION
**Pull request by Anselmoo**
Additional refactors based on the PR #2206

### CODE REVIEW

File: src/black/linegen.py [hunk 1]
```diff
@@ -812,35 +813,40 @@ def maybe_make_parens_invisible_in_atom(node: LN, parent: LN) -> bool:
...
-        if is_atom_with_invisible_parens(middle):
-            # Strip the invisible parens from `middle` by replacing
-            # it with the child in-between the invisible parens
-            middle.replace(middle.children[1])
+    return True
 
-        return False
 
-    return True
+def _extracted_from_maybe_make_parens_invisible_in_atom_28(node, first, last, parent):
```
Comment (JelleZijlstra) [inline src/black/linegen.py:837]: What does 28 mean here?

File: src/blib2to3/pgen2/conv.py [hunk 2]
```diff
@@ -67,12 +67,10 @@ def parse_graminit_h(self, filename):
...
         self.symbol2number = {}
         self.number2symbol = {}
-        lineno = 0
-        for line in f:
-            lineno += 1
+        for lineno, line in enumerate(f, start=1):
             mo = re.match(r"^#define\s+(\w+)\s+(\d+)$", line)
             if not mo and line.strip():
-                print("%s(%s): can't parse %s" % (filename, lineno, line.strip()))
+                print("%s(%s): can't parse %s" % (filename, lineno+1, line.strip()))
```
Comment (JelleZijlstra) [inline src/blib2to3/pgen2/conv.py:73]: You need either this or the start=1 but not both.

File: src/black/parsing.py [hunk 3]
```diff
@@ -102,8 +102,7 @@ def lib2to3_parse(src_txt: str, target_versions: Iterable[TargetVersion] = ()) -
 
 def lib2to3_unparse(node: Node) -> str:
     """Given a lib2to3 node, return its string representation."""
-    code = str(node)
-    return code
+    return str(node)
```
Comment (MarcoGorelli) [inline src/black/parsing.py:105]: just a suggestion - perhaps make separate PRs, one for anything totally non-controversial (like this), and another for things like factoring out `_extracted_from_maybe_make_parens_invisible_in_atom_28`?

Comment (Anselmoo) [reply src/black/parsing.py:105]: Sorry for the in `atom_28`, this was a passed issue!
I didn't know that you prefer separated PRs.  Go it

File: src/black/linegen.py (hunk 1 above)
Comment (Anselmoo) [reply src/black/linegen.py:837]: Sorry, the number `28` is just a filler for the help-functions, but actually not needed

File: src/blib2to3/pgen2/conv.py (hunk 2 above)
Comment (Anselmoo) [reply src/blib2to3/pgen2/conv.py:73]: Yes, you are right!

File: src/black/parsing.py (hunk 3 above)
Comment (MarcoGorelli) [reply src/black/parsing.py:105]: I'm not a maintainer so it's not my call, but in general small and focused PRs are easier to review
